generator/integration_tests/generator_integration_test.cc

```diff
@@ -0,0 +1,138 @@
+#include <iostream>
+
+namespace google {
+namespace cloud {
+namespace generator_internal {
+namespace {
+
+StatusOr<std::vector<std::string>> ReadFile(std::string const& filepath) {
+  std::string line;
+  std::vector<std::string> file_contents;
+  std::ifstream input_file(filepath);
+
+  if (!input_file)
+    return Status(StatusCode::kNotFound, "Cannot open: " + filepath);
+  while (std::getline(input_file, line)) {
+    file_contents.push_back(line);
+  }
+  return file_contents;
+}
+
+class GeneratorIntegrationTest : public testing::Test {
+ protected:
+  void SetUp() override {
+    auto run_integration_tests =
+        google::cloud::internal::GetEnv(
+            "GOOGLE_CLOUD_CPP_GENERATOR_RUN_INTEGRATION_TESTS")
+            .value_or("");
+    if (run_integration_tests != "yes") {
+      GTEST_SKIP();
+    }
+
+    EXPECT_EQ(run_integration_tests, "yes");
+    ASSERT_TRUE(
+        google::cloud::internal::GetEnv("GOOGLE_CLOUD_CPP_GENERATOR_PROTO_PATH")
+            .has_value());
+    ASSERT_TRUE(google::cloud::internal::GetEnv(
+                    "GOOGLE_CLOUD_CPP_GENERATOR_GOOGLEAPIS_PATH")
+                    .has_value());
+
+    // Path to find .proto files distributed with protobuf.
+    auto proto_path =
+        google::cloud::internal::GetEnv("GOOGLE_CLOUD_CPP_GENERATOR_PROTO_PATH")
+            .value();
+
+    // Path to find .proto files distributed with googleapis/googleapis repo.
+    auto googleapis_path = google::cloud::internal::GetEnv(
+                               "GOOGLE_CLOUD_CPP_GENERATOR_GOOGLEAPIS_PATH")
+                               .value();
+
+    // Path to find .proto files defined for these tests.
+    auto code_path =
+        google::cloud::internal::GetEnv("GOOGLE_CLOUD_CPP_GENERATOR_CODE_PATH")
+            .value_or("/v");
+
+    // Path to location where generated code is written.
+    auto output_path = google::cloud::internal::GetEnv(
+                           "GOOGLE_CLOUD_CPP_GENERATOR_OUTPUT_PATH")
+                           .value_or("/tmp");
+
+    google::cloud::generator::Generator generator;
+    //    google::protobuf::compiler::CommandLineInterface cli;
+    cli_.RegisterGenerator("--cpp_codegen_out", "--cpp_codegen_opt", &generator,
+                           "Codegen C++ Generator");
+
+    std::vector<std::string> args;
+    // empty arg keeps first real arg from being ignored.
+    args.emplace_back("");
+    args.emplace_back("--proto_path=" + proto_path);
+    args.emplace_back("--proto_path=" + googleapis_path);
+    args.emplace_back("--proto_path=" + code_path);
+    args.emplace_back("--cpp_codegen_out=" + output_path);
+    args.emplace_back("--cpp_codegen_opt=product_path=google/cloud/test/");
+    args.emplace_back("generator/integration_tests/test.proto");
+
+    c_args_.reserve(args.size());
+    for (auto const& arg : args) {
+      std::cout << "args : " << arg << std::endl;
+      c_args_.push_back(arg.c_str());
+    }
+  }
+
+  int GenerateOnce() {
```

Comment:
Consider inlining the body of this function in the `SetUp` function. I think that may simplify the code and semantics. That will also let you get rid of the data members in this fixture, which is nice.